desktop/core/src/desktop/templates/login.mako

```diff
@@ -107,7 +107,7 @@ ${ commonheader(_("Welcome to Hue"), "login", user, request, "50px", True, True)
         % if 'AllowAllBackend' in backend_names or backend_names == ['OAuthBackend']:
           hide
         % endif
-        % if form['password'].errors or (login_errors and ('username' in form and not form['username'].errors) and not form['password'].errors):
+        % if form['password'].errors or (login_errors and not form['username'].errors and not form['password'].errors):
```

Comment:
Strange that `'username' in form` does not work, what would be a better way to check that form['username'] does not fail if the Authentication backend does not have a username? (e.g. some backend don't have 'username')

Reply:
> Strange that `'username' in form` does not work, what would be a better way to check that form['username'] does not fail if the Authentication backend does not have a username? (e.g. some backend don't have 'username')
okay I'll check it.

Reply:
AuthenticationForm have username as BoundField, but `'username' in auth_form` is False.
In case, `auth_form` which doensn't have value for `'username'`
```
[10/Jun/2020 21:33:42 +0000] views        INFO     type(auth_form)
[10/Jun/2020 21:33:42 +0000] views        INFO     <class 'desktop.auth.forms.AuthenticationForm'>

[10/Jun/2020 21:33:42 +0000] views        INFO     type(auth_form['username'])
[10/Jun/2020 21:33:42 +0000] views        INFO     <class 'django.forms.boundfield.BoundField'>

[10/Jun/2020 21:33:42 +0000] views        INFO     'username' in auth_form
[10/Jun/2020 21:33:42 +0000] views        INFO     False

```

Reply:
I fixed it `'username' in form.fields`